jsonnet/control-plane.libsonnet

```diff
@@ -24,6 +24,22 @@ function(params)
       },
     },
 
+    prometheusRule+: {
+      spec+: {
+        groups+: [{
+          name: 'openshift-dc.rules',
+          rules: [
+            {
+              expr:
+                'max by (cluster, namespace, workload, pod) (label_replace(label_replace(kube_pod_owner{job="kube-state-metrics", owner_kind="ReplicationController"},"replicationcontroller", "$1", "owner_name", "(.*)") * on(replicationcontroller, namespace) group_left(owner_name) topk by(replicationcontroller, namespace) (1, max by (replicationcontroller, namespace, owner_name) (kube_replicationcontroller_owner{job="kube-state-metrics"})),"workload", "$1", "owner_name", "(.*)"))',
+              labels: { workload_type: 'deployment-config' },
+              record: 'namespace_workload_pod:kube_pod_owner:relabel',
+            },
+          ],
+        }],
+      },
+    },
```

Comment:
We could patch the existing `k8s.rules` group so that the new rule is located with the other `namespace_workload_pod:kube_pod_owner:relabel` rules.

```
    prometheusRule+: {
      spec+: {
        groups: [
          if g.name == 'k8s.rules' then
            g {
              rules+: [
                {
                  expr:
                    'max by (cluster, namespace, workload, pod) (label_replace(label_replace(kube_pod_owner{job="kube-state-metrics", owner_kind="ReplicationController"},"replicationcontroller", "$1", "owner_name", "(.*)") * on(replicationcontroller, namespace) group_left(owner_name) topk by(replicationcontroller, namespace) (1, max by (replicationcontroller, namespace, owner_name) (kube_replicationcontroller_owner{job="kube-state-metrics"})),"workload", "$1", "owner_name", "(.*)"))',
                  labels: { workload_type: 'deployment-config' },
                  record: 'namespace_workload_pod:kube_pod_owner:relabel',
                },
              ],
            }
          else
            g
          for g in super.groups
        ],
```

Reply:
I did not do this because I felt that deployment-config is not a k8s object.